### PR TITLE
tests: Add a new benchmark test for multicore idle

### PR DESCRIPTION
**Pull request by carlescufi**
This brings back the multicore hello_world sample removed in c920ae05bd19e2d9f4921aa4df73d54e3fe0fd4e, but this time as a benchmark test that can be used as a means to measure power consumption in multicore devices.

### CODE REVIEW

**Review by katgiadla — CHANGES_REQUESTED** [review]
Please add responsible person for these tests to file `CODEOWNERS`.

File: tests/benchmarks/multicore/power_management/Kconfig
Comment (gmarull) [inline tests/benchmarks/multicore/power_management/Kconfig:12]: nRF53 does not support PM, so it's pointless here.

File: tests/benchmarks/multicore/power_management/src/main.c
Comment (gmarull) [inline tests/benchmarks/multicore/power_management/src/main.c:17]: logging is irrelevant for PM demo

**Review by gmarull — CHANGES_REQUESTED** [review]
needs updates (eg nRF53 is pointless here)

File: tests/benchmarks/multicore/power_management/src/main.c
Comment (carlescufi) [reply tests/benchmarks/multicore/power_management/src/main.c:17]: But we may want to print the iterations, do you suggest `printk()` ?

**Comment by gmarull** [comment]
> The name can be a bit confusing. I recommend to change it from `multicore/power_management` to `multicore/idle`. It is used to check basic multicore stability on nRF53 as well so it should be kept for this platform

For nRF53 there’s already a sample upstream, so there’s no need to keep duplicates.

File: tests/benchmarks/multicore/power_management/src/main.c
Comment (kl-cruz) [reply tests/benchmarks/multicore/power_management/src/main.c:17]: Logging is important here, to do basic check if variables/memories are restored correctly after execution ksleep and landing in the idle/sleep mode. Additionaly incrementation of this variable is kind of "heart beat" feature that showing us that CPU is in correct place in case of code execution, escpecially after ksleep and for example does not reset completely.

Comment (gmarull) [reply tests/benchmarks/multicore/power_management/src/main.c:17]: This can be printed together with the printk below, there's no need to add an extra variable to the game (logging subsys). Also, continued printing with a counter can be added to the upstream sample if needed, so that no special code is needed, at least for nRF53. And, for nRF54 we should not need anything else other than `CONFIG_PM=y` and potentially a configurable sleep period over Kconfig so multiple idle states can be easily tested.

**Comment by carlescufi** [comment]
Discussions resolved offline

File: tests/benchmarks/multicore/idle/README.rst
Comment (gmarull) [inline tests/benchmarks/multicore/idle/README.rst:102]: messages need update

File: CODEOWNERS
Comment (kl-cruz) [inline CODEOWNERS:296]: We are not owners of this space, we only use it

File: tests/benchmarks/multicore/idle/remote/prj.conf
Comment (gmarull) [inline tests/benchmarks/multicore/idle/remote/prj.conf:1]:
```suggestion
CONFIG_PRINTK=y
CONFIG_LOG=y
```

**Review by kl-cruz — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by nordic-segl — APPROVED** [review]

File: tests/benchmarks/multicore/idle/src/main.c
Comment (adamkondraciuk) [inline tests/benchmarks/multicore/idle/src/main.c:19]: Do we need `\n` here? The effect is double interline.

Comment (carlescufi) [reply tests/benchmarks/multicore/idle/src/main.c]: Fixed

**Review by ns-tolu — APPROVED** [review]

**Review by adamkondraciuk — APPROVED** [review]

File: tests/benchmarks/multicore/idle/testcase.yaml
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle/testcase.yaml:2]: Please do not use build_only,
rather define expected regex on console, eg:
  harness: console
  harness_config:
    type: multi_line
    ordered: true
    regex:
      - "Multicore idle test on"
      - "Multicore idle test iteration 0"
      - "Multicore idle test iteration 1"

**Comment by greg-fer** [comment]
Reviewing docs now.

File: tests/benchmarks/multicore/idle/README.rst
Comment (greg-fer) [inline tests/benchmarks/multicore/idle/README.rst:26]:
```suggestion
The test demonstrates how to build a multicore idle application with :ref:`configuration_system_overview_sysbuild`.
```

File: tests/benchmarks/multicore/idle/README.rst
Comment (greg-fer) [inline tests/benchmarks/multicore/idle/README.rst:27]:
```suggestion
When building with sysbuild, the build system adds child images based on the options selected in the project's additional configuration and build files.
```

File: tests/benchmarks/multicore/idle/README.rst
Comment (greg-fer) [inline tests/benchmarks/multicore/idle/README.rst:32]:
```suggestion
* :file:`Kconfig.sysbuild` - This file is used to add :ref:`sysbuild Kconfig options <configuration_system_overview_sysbuild>` that are passed to all the images.
```

File: tests/benchmarks/multicore/idle/README.rst
Comment (greg-fer) [inline tests/benchmarks/multicore/idle/README.rst:68]:
```suggestion
  .. note::
    
     :ref:`zephyr:nordic-ppr` is used in the configuration above to automatically launch PPR core from the application core.
```

File: tests/benchmarks/multicore/idle/README.rst
Comment (greg-fer) [inline tests/benchmarks/multicore/idle/README.rst:70]:
```suggestion
  An additional configuration setup is provided to execute code directly from the non-volatile memory (MRAM) on the PPR core.
```

File: tests/benchmarks/multicore/idle/README.rst
Comment (greg-fer) [inline tests/benchmarks/multicore/idle/README.rst:87]: Indent not needed.
```suggestion
1. |connect_terminal|
#. Reset the kit.
#. Observe the console output for both cores:
```
Please update the lines that follow accordingly.

**Review by greg-fer — CHANGES_REQUESTED** [review]
I think a changelog entry is needed for the addition of the new software component.

**Comment by carlescufi** [comment]
> I think a changelog entry is needed for the addition of the new software component.

This is really just a test for internal use @b-gent. I don't think it is needed.

**Review by nordic-piks — APPROVED** [review]

File: tests/benchmarks/multicore/idle/README.rst
Comment (greg-fer) [inline tests/benchmarks/multicore/idle/README.rst:98]:
```suggestion
   * For the application core, the output should be as follows:

     .. code-block:: console

       *** Booting nRF Connect SDK zephyr-v3.5.0-3517-g9458a1aaf744 ***
       Multi-core idle test on nrf5340dk/nrf5340/cpuapp
       Multi-core idle test iteration 0
       Multi-core idle test iteration 1
       ...
```

File: tests/benchmarks/multicore/idle/README.rst
Comment (greg-fer) [inline tests/benchmarks/multicore/idle/README.rst:108]:
```suggestion
   * For the remote core, the output should be as follows:

     .. code-block:: console

        *** Booting nRF Connect SDK zephyr-v3.5.0-3517-g9458a1aaf744 ***
        Multi-core idle test on nrf5340dk/nrf5340/cpunet
        Multi-core idle test iteration 0
        Multi-core idle test iteration 1
        ...
```

**Review by greg-fer — APPROVED** [review]
Approving, though one part of one comment was not applied, hence two more proposed edits.

**Comment by carlescufi** [comment]
> I think a changelog entry is needed for the addition of the new software component.

Applied the rest

**Comment by b-gent** [comment]
> > I think a changelog entry is needed for the addition of the new software component.
> 
> This is really just a test for internal use @b-gent. I don't think it is needed.

no changelog is fine

**Review by b-gent — APPROVED** [review]

**Pull request merged**
